app/controllers/spree/admin/payment_methods_controller.rb

```diff
@@ -197,10 +197,16 @@ def update_params
 
       def clear_preference_cache
         @payment_method.calculator.preferences.each_key do |key|
-          Rails.cache.delete(@payment_method.calculator.preference_cache_key(key))
+          delete_from_cache @payment_method.calculator.preference_cache_key(key)
         end
       end
 
+      def delete_from_cache(cache_key)
+        return unless cache_key.class.in? [String, Integer, Float, Symbol]
+
+        Rails.cache.delete(cache_key)
+      end
+
```

Comment:
I wonder if this will still be needed when hiredis is added again.

Reply:
This is marked as resolved, so I guess you found that it _was_ still needed when using hiredis-client?

Reply:
Yes, I found that this was still needed.